apps/main/src/dao/components/ConnectEthereum.tsx

```diff
@@ -1,22 +1,17 @@
 import { useRouter } from 'next/navigation'
-import { CONNECT_STAGE } from '@/dao/constants'
-import useStore from '@/dao/store/useStore'
 import { getPath, getRestFullPathname } from '@/dao/utils/utilsRouter'
 import Button from '@ui/Button'
+import { useConnection } from '@ui-kit/features/connect-wallet'
 import { t } from '@ui-kit/lib/i18n'
 
 export const ConnectEthereum = () => {
-  const connectState = useStore((state) => state.connectState)
-  const updateConnectState = useStore((state) => state.updateConnectState)
-
+  const { connectState } = useConnection()
   const { push } = useRouter()
-  const restPartialPathname = getRestFullPathname()
-  const isLoading = connectState.status === 'loading'
-
-  const handleSwitchNetwork = () => {
-    push(getPath({ network: 'ethereum' }, `/${restPartialPathname}`))
-    updateConnectState('loading', CONNECT_STAGE.SWITCH_NETWORK, [1, 1])
-  }
-
-  return <Button variant="filled" onClick={handleSwitchNetwork} loading={isLoading}>{t`Connect to Ethereum`}</Button>
+  return (
+    <Button
+      variant="filled"
+      onClick={() => push(getPath({ network: 'ethereum' }, `/${getRestFullPathname()}`))}
+      loading={connectState.status === 'loading'}
```

Comment:
no `isLoading()` call from `ConnectionContext.tsx?`. It boils down to the same code so it doesn't matter too much